File: _c-sharp-corner/Multithreading with C#/UsingDedicatedThreads.cs
using System;
using System.Threading;

class UsingDedicatedThreads
{
    static void Main(string[] args)
    {
        // create thread start delegate instance - contains the method to execute by the thread
        ThreadStart ts = new ThreadStart(run);
        // create new thread
        Thread thrd = new Thread(ts);
        // start thread
        thrd.Start();

        // makes the main thread sleep - let sub thread to run
        Thread.Sleep(1000);

        for (int t = 10; t > 0; t--)
        {
            Console.WriteLine("Main Thread value is :" + t);
            Thread.Sleep(1000);
        }

        Console.WriteLine("Good Bye!!!I'm main Thread");
        Console.ReadLine();

    }

    // this method executed by a separate thread
    static void run()
    {
        for (int i = 0; i < 10; i++)
        {
            Console.WriteLine("Sub Thread value is : " + i);
            Thread.Sleep(1000);
        }
        Console.WriteLine("Good Bye!!!I'm Sub Thread");
    }
}

File: _c-sharp-corner/Multithreading with C#/UsingDedicatedThreads.vb
Imports System.Threading

Module UsingDedicatedThreads

    Sub Main()

        'create thread start delegate instance - contains the method to execute by the thread
        Dim ts As ThreadStart = New ThreadStart(AddressOf run)
        ' create new thread
        Dim t As Thread
        t = New Thread(ts)
        ' start thread
        t.Start()

        ' makes the main thread sleep - let sub thread to run
        Thread.Sleep(1000)

        For i As Integer = 10 To 0 Step -1
            Console.WriteLine("Main thread value is" + i.ToString())
            Thread.Sleep(1000)
        Next i

        Console.WriteLine("Good Bye!!!I'm main Thread")
        Console.ReadKey()

    End Sub

    ' this method executed by a separate thread
    Sub run()

        For i As Integer = 0 To 10
            Console.WriteLine("Sub thread value is" + i.ToString())
            Thread.Sleep(1000)
        Next i

        Console.WriteLine("Good Bye!!!I'm Sub Thread")
    End Sub

End Module
